osmomath/decimal.go

```diff
@@ -494,6 +496,27 @@ func (d BigDec) MustFloat64() float64 {
 	}
 }
 
+// SdkDec returns the Sdk.Dec representation of a BigDec.
+// Values in any additional decimal places are truncated.
+func (d BigDec) SdkDec() sdk.Dec {
+	precisionDiff := Precision - sdk.Precision
+	precisionFactor := new(big.Int).Exp(big.NewInt(10), big.NewInt(int64(precisionDiff)), nil)
+
+	if precisionDiff < 0 {
+		panic("invalid decimal precision")
+	}
+
+	truncatedDec := sdk.NewDecFromBigIntWithPrec(new(big.Int).Quo(d.BigInt(), precisionFactor), sdk.Precision)
```

Comment:
Yes, this is something we need to discuss, in addition to the potential issues related to the ultimate conversion back to SDK dec at the keeper level (since even though stableswap calcs use `BigDec` the pool interface remains unchanged with sdk.Dec)

The approach I've implemented here basically truncates any decimal places that go beyond sdk.Dec's precision (e.g. if `BigDec` has 19 decimal precision and `sdk.Dec` has 18, the 19th decimal is truncated). I defaulted to this to avoid having the pool send more tokens out than it should, but I'm not convinced this is a robust solution

Reply:
I think truncation is the right thing to do as well! Its just not obvious to me from the code + test (perhaps I'm missing something though)

Are we relying on big.Int Quo to be floor division? (If so we should add a comment for this. And move it into a couple lines)

Reply:
Yes, we do rely on `big.Int` Quo being floor div (which it is) – I'll add a comment and spread the calculation out to be more readable!